src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsContainedLanguageComponentsFactoryBase.cs

```diff
@@ -49,6 +67,8 @@ internal abstract class VsContainedLanguageComponentsFactoryBase : IVsContainedL
             {
                 await ProjectServices.ThreadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                await InitializeAsync().ConfigureAwait(true);
```

Comment:
Yes, thats a nice approach. I pushed the switch to UI thread further down into LS host initialization and the test passes now.